### ISSUE TITLE
Problem updating tables with binary data containing brackets.

### ISSUE THREAD

**Issue by jean343** [sensedeep/dynamodb-onetable#267]
**ISSUES will not receive priority that do not follow this template or cannot be easily reproduced or diagnosed**

**Describe the bug**
I am using the `update` command to set some Binary data to a field, and depending on the content, the update may fail. After debugging, it turned out that my binary data contained the `{` and `}` characters, and would trigger this regex `value.toString().match(/\${.*?}|@{.*?}|{.*?}/)`

**To Reproduce**

```javascript
    const schema = {
      format: "onetable:1.1.0",
      version: "0.0.1",
      indexes: {
        primary: { hash: "pk", sort: "sk" },
      },
      models: {
        Test: {
          pk: { type: String, value: "test:" },
          sk: { type: String, value: "test:" },
          binaryData: { type: "binary" },
        },
      },
    };
    const table = new Table({
      client,
      name: "test",
      schema,
    });
    const Test = table.getModel("Test");
    Test.update({}, { set: { binaryData: Buffer.from(`‰PNG   IHDR  {}`) } });
```

```
UnhandledPromiseRejectionWarning: OneTableError: OneTable execute failed "update" for "Test. Invalid UpdateExpression: Syntax error; token: "‰", near: "= ‰PNG"
```

**Expected behavior**
The binary data should be send to DynamoDB.

**Environment (please complete the following information):**
 - OS - Windows
 - Node Version - 14.18.3
 - OneTable Version - 2.3.1

**Sample fix**
A simple fix would be to check for Buffers, or to change the regex.
```javascript
if (!Buffer.isBuffer(value) && value.toString().match(/\${.*?}|@{.*?}|{.*?}/)) {
```

**Comment by mobsense** [comment]
Here is a patch to correct. Can you please test if this fixes the issue for you?

```
diff --git a/src/Expression.js b/src/Expression.js
index bc35790..2e6437b 100644
--- a/src/Expression.js
+++ b/src/Expression.js
@@ -386,8 +386,7 @@ export class Expression {
                 }
                 this.already[key] = true
                 let target = this.makeTarget(fields, key)
-                //  If value is number of simple string then don't expand
-                if (value.toString().match(/\${.*?}|@{.*?}|{.*?}/)) {
+                if (typeof value == 'string' && value.match(/\${.*?}|@{.*?}|{.*?}/)) {
                     updates.set.push(`${target} = ${this.expand(value)}`)
                 } else {
                     updates.set.push(`${target} = :_${this.addValue(value)}`)
```

**Comment by jean343** [comment]
Thank you so much for the quick reply!

I just tested your patch, and it works perfectly. I like the `typeof` more than the `isBuffer` check.

**Comment by mobsense** [comment]
That is now committed and will be released in 2.3.2 next week.
